language-support/java/bindings-rxjava/src/main/java/com/daml/ledger/rxjava/grpc/CommandsBuilder.java

```diff
@@ -0,0 +1,246 @@
+    return actAs;
+  }
+
+  public List<String> getReadAs() {
+    return readAs;
+  }
+
+  public Optional<Instant> getMinLedgerTimeAbs() {
+    return minLedgerTimeAbs;
+  }
+
+  public Optional<Duration> getMinLedgerTimeRel() {
+    return minLedgerTimeRel;
+  }
+
+  public Optional<Duration> getDeduplicationTime() {
+    return deduplicationTime;
+  }
+
+  public List<? extends HasCommands> getCommands() {
+    return commands;
+  }
+
+  public Optional<String> getAccessToken() {
+    return accessToken;
+  }
+
+  public CommandsBuilder withWorkflowId(String workflowId) {
+    return new CommandsBuilder(
+        workflowId,
+        applicationId,
+        commandId,
+        actAs,
+        readAs,
+        minLedgerTimeAbs,
+        minLedgerTimeRel,
+        deduplicationTime,
+        commands,
+        accessToken);
+  }
+
+  public CommandsBuilder withParty(String party) {
+    return new CommandsBuilder(
+        workflowId,
+        applicationId,
+        commandId,
+        singletonList(party),
+        readAs,
+        minLedgerTimeAbs,
+        minLedgerTimeRel,
+        deduplicationTime,
+        commands,
+        accessToken);
+  }
+
+  public CommandsBuilder withActAs(List<@NonNull String> actAs) {
+    return new CommandsBuilder(
+        workflowId,
+        applicationId,
+        commandId,
+        actAs,
+        readAs,
+        minLedgerTimeAbs,
+        minLedgerTimeRel,
+        deduplicationTime,
+        commands,
+        accessToken);
+  }
+
+  public CommandsBuilder withReadAs(List<@NonNull String> readAs) {
+    return new CommandsBuilder(
+        workflowId,
+        applicationId,
+        commandId,
+        actAs,
+        readAs,
+        minLedgerTimeAbs,
+        minLedgerTimeRel,
+        deduplicationTime,
+        commands,
+        accessToken);
+  }
+
+  public CommandsBuilder withMinLedgerTimeAbs(Optional<Instant> minLedgerTimeAbs) {
+    return new CommandsBuilder(
+        workflowId,
+        applicationId,
+        commandId,
+        actAs,
+        readAs,
+        minLedgerTimeAbs,
+        minLedgerTimeRel,
+        deduplicationTime,
+        commands,
+        accessToken);
+  }
+
+  public CommandsBuilder withMinLedgerTimeRel(Optional<Duration> minLedgerTimeRel) {
+    return new CommandsBuilder(
+        workflowId,
+        applicationId,
+        commandId,
+        actAs,
+        readAs,
+        minLedgerTimeAbs,
+        minLedgerTimeRel,
+        deduplicationTime,
+        commands,
+        accessToken);
+  }
+
+  public CommandsBuilder withDeduplicationTime(Optional<Duration> deduplicationTime) {
+    return new CommandsBuilder(
+        workflowId,
+        applicationId,
+        commandId,
+        actAs,
+        readAs,
+        minLedgerTimeAbs,
+        minLedgerTimeRel,
+        deduplicationTime,
+        commands,
+        accessToken);
+  }
+
+  public CommandsBuilder withCommands(List<@NonNull ? extends HasCommands> commands) {
```

Comment:
Do we want to do a union here?
```java
List<? extends HasCommands> newCommands = Stream.concat(this.commands.stream(), commands.stream()).collect(toList());

return new CommandsBuilder(
         workflowId,
         applicationId,
         commandId,
         actAs,
         readAs,
         minLedgerTimeAbs,
         minLedgerTimeRel,
         deduplicationTime,
         newCommands,
         accessToken);
```

Reply:
`withCommands` should just replace the commands, if users want to do a union or anything else they can do that themselves.

Reply:
(if I understand correctly)